test/unit/query-generator/json-path-extraction-query.test.ts

```diff
@@ -0,0 +1,39 @@
+import { expectsql, getTestDialect, sequelize } from '../../support';
+
+const dialectName = getTestDialect();
+
+const notSupportedError = new Error(`JSON operations are not supported in ${dialectName}.`);
+
+describe('QueryGenerator#jsonPathExtractionQuery', () => {
+  const queryGenerator = sequelize.getQueryInterface().queryGenerator;
+
+  it('should handle isJson parameter true', () => {
+    expectsql(() => queryGenerator.jsonPathExtractionQuery('profile', 'id', true), {
+      default: notSupportedError,
+      mariadb: 'json_unquote(json_extract(`profile`,\'$.id\'))',
+      mysql: 'json_unquote(json_extract(`profile`,\'$.\\"id\\"\'))',
+      postgres: `("profile"#>'{id}')`,
+      sqlite: 'json_extract(`profile`,\'$.id\')',
+    });
+  });
+
+  it('should use default handling if isJson is false', () => {
+    expectsql(() => queryGenerator.jsonPathExtractionQuery('profile', 'id', false), {
+      default: notSupportedError,
+      mariadb: 'json_unquote(json_extract(`profile`,\'$.id\'))',
+      mysql: 'json_unquote(json_extract(`profile`,\'$.\\"id\\"\'))',
+      postgres: `("profile"#>>'{id}')`,
+      sqlite: 'json_extract(`profile`,\'$.id\')',
+    });
+  });
+
+  it('should use default handling if isJson is not passed', () => {
+    expectsql(() => queryGenerator.jsonPathExtractionQuery('profile', 'id'), {
+      default: notSupportedError,
+      mariadb: 'json_unquote(json_extract(`profile`,\'$.id\'))',
+      mysql: 'json_unquote(json_extract(`profile`,\'$.\\"id\\"\'))',
+      postgres: `("profile"#>>'{id}')`,
```

Comment:
we have too few tests for the path here. What happens if one of the entries in the path includes characters such as `'`, `}`, `,`, or `{`?

`'{id}'` is just an array. IMO we need to drop the [custom array building we have here](https://github.com/sequelize/sequelize/blob/main/src/dialects/postgres/query-generator.js#L1038) and use the stringifier implemented in the Array DataType. Having two separate implementations is going to lead to bugs

(also doesn't need to be implemented in this PR, can be a TODO)